src/category_theory/abelian/homology.lean

```diff
@@ -0,0 +1,255 @@
+  See `homology_iso_cokernel_lift`. -/
+abbreviation homology_c : A :=
+cokernel (kernel.lift g f w)
+
+/-- The kernel of `cokernel.desc f g w`. This is isomorphic to `homology f g w`.
+  See `homology_iso_kernel_desc`. -/
+abbreviation homology_k : A :=
+kernel (cokernel.desc f g w)
+
+/-- The canonical map from `homology_c` to `homology_k`.
+  This is an isomorphism, and it is used in obtaining the API for `homology f g w`
+  in the bottom of this file. -/
+abbreviation homology_c_to_k : homology_c f g w ⟶ homology_k f g w :=
+cokernel.desc _ (kernel.lift _ (kernel.ι _ ≫ cokernel.π _) (by simp)) begin
+  apply limits.equalizer.hom_ext,
+  simp,
+end
+
+local attribute [instance] pseudoelement.hom_to_fun pseudoelement.has_zero
+
+instance : mono (homology_c_to_k f g w) :=
+begin
+  apply pseudoelement.mono_of_zero_of_map_zero,
+  intros a ha,
+  obtain ⟨a,rfl⟩ := pseudoelement.pseudo_surjective_of_epi (cokernel.π (kernel.lift g f w)) a,
+  apply_fun (kernel.ι (cokernel.desc f g w)) at ha,
+  simp only [←pseudoelement.comp_apply, cokernel.π_desc,
+    kernel.lift_ι, pseudoelement.apply_zero] at ha,
+  simp only [pseudoelement.comp_apply] at ha,
+  haveI : exact f (cokernel.π f) := exact_cokernel f,
+  obtain ⟨b,hb⟩ : ∃ b, f b = _ := pseudoelement.pseudo_exact_of_exact.2 _ ha,
+  suffices : ∃ c, kernel.lift g f w c = a,
+  { obtain ⟨c,rfl⟩ := this,
+    simp [← pseudoelement.comp_apply] },
+  use b,
+  apply_fun kernel.ι g,
+  swap, { apply pseudoelement.pseudo_injective_of_mono },
+  simpa [← pseudoelement.comp_apply]
+end
+
+instance : epi (homology_c_to_k f g w) :=
+begin
+  apply pseudoelement.epi_of_pseudo_surjective,
+  intros a,
+  let b := kernel.ι (cokernel.desc f g w) a,
+  haveI : exact f (cokernel.π f) := exact_cokernel f,
+  obtain ⟨c,hc⟩ : ∃ c, cokernel.π f c = b,
+    apply pseudoelement.pseudo_surjective_of_epi (cokernel.π f),
+  have : g c = 0,
+  { dsimp [b] at hc,
+    rw [(show g = cokernel.π f ≫ cokernel.desc f g w, by simp), pseudoelement.comp_apply, hc],
+    simp [← pseudoelement.comp_apply] },
+  obtain ⟨d,hd⟩ : ∃ d, kernel.ι g d = c,
+  { apply pseudoelement.pseudo_exact_of_exact.2 _ this,
+    apply exact_kernel_ι },
+  use cokernel.π (kernel.lift g f w) d,
+  apply_fun kernel.ι (cokernel.desc f g w),
+  swap, { apply pseudoelement.pseudo_injective_of_mono },
+  simp only [←pseudoelement.comp_apply, cokernel.π_desc, kernel.lift_ι],
+  simp only [pseudoelement.comp_apply, hd, hc],
+end
+
+instance (w : f ≫ g = 0) : is_iso (homology_c_to_k f g w) := is_iso_of_mono_of_epi _
+
+end category_theory.abelian
+
+/-- The homology associated to `f` and `g` is isomorphic to a kernel. -/
+def homology_iso_kernel_desc : homology f g w ≅ kernel (cokernel.desc f g w) :=
+homology_iso_cokernel_lift _ _ _ ≪≫ as_iso (category_theory.abelian.homology_c_to_k _ _ _)
+
+namespace homology
+
+-- `homology.π` is taken
+/-- The canonical map from the kernel of `g` to the homology of `f` and `g`. -/
+def π' : kernel g ⟶ homology f g w :=
```

Comment:
Maybe we should change the name of `homology.π`? Since it goes *to* the homology it seems wrong to me.

Reply:
This is meant to be analogous to `cokernel.\pi`, i.e. the projection onto the cokernel. The homology is the quotient of the kernel, so I think `\pi` makes sense.

Reply:
Ah, you're right.